File: app/src/main/java/com/past/music/entity/FolderEntity.java
package com.past.music.entity;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/3/4 下午6:20
 * 描述：音频文件的文件夹信息
 * 备注：Copyright © 2010-2017. gaojin All rights reserved.
 * =======================================================
 */
public class FolderEntity implements Parcelable {

    private String folder_name;
    private String folder_path;
    private String folder_sort;
    private int folder_count;

    public FolderEntity() {
    }

    protected FolderEntity(Parcel in) {
        folder_name = in.readString();
        folder_path = in.readString();
        folder_sort = in.readString();
        folder_count = in.readInt();
    }

    public static final Creator<FolderEntity> CREATOR = new Creator<FolderEntity>() {
        @Override
        public FolderEntity createFromParcel(Parcel in) {
            return new FolderEntity(in);
        }

        @Override
        public FolderEntity[] newArray(int size) {
            return new FolderEntity[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(folder_name);
        dest.writeString(folder_path);
        dest.writeString(folder_sort);
        dest.writeInt(folder_count);
    }

    public String getFolder_name() {
        return folder_name;
    }

    public void setFolder_name(String folder_name) {
        this.folder_name = folder_name;
    }

    public String getFolder_path() {
        return folder_path;
    }

    public void setFolder_path(String folder_path) {
        this.folder_path = folder_path;
    }

    public String getFolder_sort() {
        return folder_sort;
    }

    public void setFolder_sort(String folder_sort) {
        this.folder_sort = folder_sort;
    }

    public int getFolder_count() {
        return folder_count;
    }

    public void setFolder_count(int folder_count) {
        this.folder_count = folder_count;
    }
}

File: app/src/main/java/com/se/music/common/entity/FolderEntity.kt
package com.se.music.common.entity

import android.os.Parcel
import android.os.Parcelable

/**
 *Author: gaojin
 *Time: 2018/5/13 下午4:27
 */

class FolderEntity() : Parcelable {
    var folder_name: String? = null
    var folder_path: String? = null
    var folder_sort: String? = null
    var folder_count: Int = 0

    constructor(parcel: Parcel) : this() {
        folder_name = parcel.readString()
        folder_path = parcel.readString()
        folder_sort = parcel.readString()
        folder_count = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(folder_name)
        parcel.writeString(folder_path)
        parcel.writeString(folder_sort)
        parcel.writeInt(folder_count)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<FolderEntity> {
        override fun createFromParcel(parcel: Parcel): FolderEntity {
            return FolderEntity(parcel)
        }

        override fun newArray(size: Int): Array<FolderEntity?> {
            return arrayOfNulls(size)
        }
    }
}